deps/corepack/dist/npm.js

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
```

Comment:
these files could be symlinks to a helper that reads process.execName or whatever it's called. it's a common pattern in linux core utils so it shouldn't break anything.

Reply:
it also occurs to me that if someone does not want a specific package manager on their system (for example, i do not wish to have yarn on my system), this could be considered somewhat invasive. Are we planning to ship tars with and without corepack?

Reply:
@devsnek ... this would not install the binaries by default. It installs jumpers that would download the binaries on demand if needed and it would be possible to block it via configuration.

Reply:
What if it will prompt you before downloading yarn or pnpm?

Reply:
It already prompts you asking which version you want to use, if you answer no it wont download anything. The name `yarn` and `yarnpkg` is still in the path though

Reply:
We will need to test the impact it will have on the ecosystem. There are tools that behave differently if `yarn` is present.
For example, Create React App uses `yarn` instead of `npm` ["when available"](https://create-react-app.dev/docs/getting-started/#selecting-a-package-manager) (I don't know what "available" means exactly).

Reply:
How would tools even detect “the user explicitly made yarn available” without the default case being “the yarn binary is not available”?

Reply:
Ideally tools should check `process.env.npm_config_user_agent` to see which package manager invoked them and use that

Reply:
I'm not sure if it's implemented yet or not, but there was also discussion around being able to set configuration options to disable the download. It would likely be possible to have the tool remove the jumpers from the path when requested to do so. For interactive installers, it should be possible to make that configurable (e.g. by default install all jumpers but give the user the option to select them during the install)

Reply:
Downloads can be disabled by setting `COREPACK_ENABLE_NETWORK=0` in the environment (https://github.com/arcanis/corepack/issues/4). Removing the jumpers isn't possible yet as I wasn't sure of the use case, although it would be possible if needed.

Reply:
@merceyz many tools are invoked directly, and not as part of a package manager - how would these tools detect it?

Reply:
I believe it should be up to these projects to make decisions on a per-case basis, as the semantic compatible path isn't clear. For instance, that CRA "will use Yarn to install dependencies (when available)" could be interpreted as CRA preferring installs to be made via Yarn as much as possible - in which case using npm when Yarn is available wouldn't be the behavior that the project would prefer.

Reply:
Also note that these tools are typically called via `npx`, `pnpx`, or `yarn create` / `yarn dlx`. In all of these cases, the tool got an implicit insight as to the preferred package manager (ie under `npx` the `$npm_config_user_agent` value will reference npm, with `yarn dlx` it'll reference Yarn, etc).